shared/src/main/java/com/couchbase/lite/Expression.java

```diff
@@ -342,48 +426,54 @@ public String toString() {
         return String.format(Locale.ENGLISH, "%s[json=%s]", getClass().getSimpleName(), asJSON());
     }
 
-    protected Object jsonValue(Object value) {
-        if (value instanceof Expression)
-            return ((Expression) value).asJSON();
-        else
-            return value;
-    }
-
     abstract Object asJSON();
 
+    static final class ValueExpression extends Expression {
+        private Object value;
+
+        ValueExpression(Object value) {
+            this.value = value;
+        }
+
+        @Override
+        Object asJSON() {
+            return this.value;
```

Comment:
If the value is Date, I understand that we need to convert to JSON date string.

Reply:
Thanks for this.